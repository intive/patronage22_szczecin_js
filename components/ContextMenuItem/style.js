import styled from 'styled-components'
import Icon from '../Icon/Icon'

export const StyledLi = styled.li`
  list-style: none;
  margin: 0;
  padding: 0;

  &:last-child button {
    margin-bottom: 0;
  }
`

export const ContextMenuButton = styled.button`
  display: flex;
  align-items: center;
  background: none;
  line-height: 1.5rem;
  color: #626262;
  border: none;
  cursor: pointer;
  padding: 0.25rem 0.5rem;
  width: 100%;
  font-size: 0.875rem;
  white-space: nowrap;

  &:hover {
    background: #bababa9e;
  }
`
export const StyledIcon = styled(Icon)`
  width: 24px;
  height: 24px;
  margin-right: 0.25rem;
  color: #7e7e7e;
`
