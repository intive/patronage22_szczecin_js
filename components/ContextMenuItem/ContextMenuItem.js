import { ContextMenuButton, StyledIcon, StyledLi } from './style'

export default function ContextMenuItem (props) {
  return (
    <StyledLi>
      <ContextMenuButton onClick={props.onClick}>
        <StyledIcon name={props.icon} />
        {props.name}
      </ContextMenuButton>
    </StyledLi>
  )
}
