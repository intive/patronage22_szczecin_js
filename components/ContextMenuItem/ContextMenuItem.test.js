/* globals describe, expect, it, jest */
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import ContextMenuItem from './ContextMenuItem'

describe('ContextMenuItem', () => {
  it('should renders correctly', () => {
    const { container } = render(<ContextMenuItem name='Testname' icon='lock' />)
    expect(container).toMatchSnapshot()
  })

  it('should call onClick prop when clicked', () => {
    const handleClick = jest.fn()

    render(
      <ContextMenuItem name='Testname' icon='lock' onClick={handleClick} />
    )
    userEvent.click(screen.getByText(/Testname/i))

    expect(handleClick).toHaveBeenCalledTimes(1)
  })
})
