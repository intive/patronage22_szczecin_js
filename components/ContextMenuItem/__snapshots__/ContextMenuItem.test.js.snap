// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContextMenuItem should renders correctly 1`] = `
<div>
  <li
    class="sc-bdvvtL jZIOSJ"
  >
    <button
      class="sc-gsDKAQ gxJInU"
    >
      <span
        class="material-icons sc-dkPtRN lhuhQn"
      >
        lock
      </span>
      Testname
    </button>
  </li>
</div>
`;
